app/src/main/java/com/github/se/gomeet/viewmodel/UserViewModel.kt

```diff
@@ -196,6 +196,37 @@ class UserViewModel(val currentUID: String? = null) : ViewModel() {
     }
   }
 
+  /**
+   * User deletes an event and removes it from their list of myEvents. It is used when a user
+   * deletes an event.
+   *
+   * @param eventId The id of the event to delete.
+   * @param userId The id of the user deleting the event.
+   */
+  suspend fun userDeletesEvent(eventId: String, userId: String = currentUID!!) {
+    try {
+      val goMeetUser = getUser(userId)!!
+      editUser(goMeetUser.copy(myEvents = goMeetUser.myEvents.minus(eventId)))
+    } catch (e: Exception) {
+      Log.w(TAG, "Couldn't delete the event", e)
+    }
+  }
+
+  /**
+   * Removes the given event from the user's list of favorites.
+   *
+   * @param eventId The id of the event to remove from favorites.
+   * @param userId The id of the user for which we remove the event from favorites.
+   */
+  suspend fun removeFavoriteEvent(eventId: String, userId: String = currentUID!!) {
```

Comment:
Again, for both of these functions it would be better (in my opinion) to just update the corresponding list rather than the whole user on Firebase, but this works too.

Reply:
Again, I don't understand what you are saying. To update the `myFavorites`field of a `GoMeetUser`, I need to update the `GoMeetUser`. If I don't then in Firebase the user would have in his `myFavorites` field, an `Event` that doesn't exist anymore if this `Event` is deleted.

Reply:
See my previous comment response about this

Reply:
In UserRepository, we update the entire user like so:
`// ...
val documentRef = Firebase.firestore.collection(USERS_COLLECTION).document(user.uid)
      documentRef
          .update(user.toMap())`

But for instance, if you have a look at the updateRating function in EventRepository, only the ratings field is updated:

`// ...
val documentRef = Firebase.firestore.collection(EVENT_COLLECTION).document(eventID)
      documentRef
          .update(RATINGS, ratings)`

Reply:
Ok my bad, I didn't see the implementation of `updateRating` function, I took inspiration from other functions in the viewModel that were implemented in the same logic that I did in this PR.